components/director/cmd/destinationfetcher-svc/main.go

```diff
@@ -0,0 +1,265 @@
+	JwksEndpoint                   string        `envconfig:"APP_JWKS_ENDPOINT,default=file://hack/default-jwks.json"`
+	DestinationsOnDemandScope      string        `envconfig:"APP_DESTINATIONS_SYNC_SCOPE,default=destinations:sync"`
+	DestinationsSensitiveDataScope string        `envconfig:"APP_DESTINATIONS_SENSITIVE_DATA_SCOPE,default=destinations_sensitive_data:read"`
+}
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	defer cancel()
+
+	term := make(chan os.Signal)
+	signal.HandleInterrupts(ctx, cancel, term)
+
+	cfg := config{}
+	err := envconfig.InitWithPrefix(&cfg, envPrefix)
+	exitOnError(err, "Error while loading app config")
+
+	ctx, err = log.Configure(ctx, &cfg.Log)
+	exitOnError(err, "Failed to configure Logger")
+
+	transactioner, closeFunc, err := persistence.Configure(ctx, cfg.Database)
+	exitOnError(err, "Error while establishing the connection to the database")
+
+	defer func() {
+		err := closeFunc()
+		exitOnError(err, "error while closing the connection to the database")
+	}()
+
+	httpClient := &http.Client{
+		Transport: httputil.NewCorrelationIDTransport(httputil.NewHTTPTransportWrapper(http.DefaultTransport.(*http.Transport))),
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+
+	destinationService := getDestinationService(cfg, transactioner)
+	handler := initAPIHandler(ctx, httpClient, cfg, destinationService)
+	runMainSrv, shutdownMainSrv := createServer(ctx, cfg, handler, "main")
+
+	go func() {
+		<-ctx.Done()
+		// Interrupt signal received - shut down the servers
+		shutdownMainSrv()
+	}()
+
+	syncJobConfig := destinationfetcher.SyncJobConfig{
+		ElectionCfg:       cfg.ElectionConfig,
+		JobSchedulePeriod: cfg.DestinationFetcherSchedule,
+		ParallelTenants:   cfg.ParallelTenantSyncs,
+	}
+	go func() {
+		err := destinationfetcher.StartDestinationFetcherSyncJob(ctx, syncJobConfig, destinationService)
+		if err != nil {
+			log.C(ctx).WithError(err).Error("Failed to start destination fetcher cronjob. Stopping app...")
+			cancel()
+		}
+	}()
+
+	runMainSrv()
+}
+
+func configureAuthMiddleware(ctx context.Context, httpClient *http.Client, router *mux.Router, cfg securityConfig, requiredScopes ...string) {
+	scopeValidator := claims.NewScopesValidator(requiredScopes)
+	middleware := authenticator.New(httpClient, cfg.JwksEndpoint, cfg.AllowJWTSigningNone, "", scopeValidator)
+	router.Use(middleware.Handler())
+
+	log.C(ctx).Infof("JWKS synchronization enabled. Sync period: %v", cfg.JWKSSyncPeriod)
+	periodicExecutor := executor.NewPeriodic(cfg.JWKSSyncPeriod, func(ctx context.Context) {
+		if err := middleware.SynchronizeJWKS(ctx); err != nil {
+			log.C(ctx).WithError(err).Errorf("An error has occurred while synchronizing JWKS: %v", err)
+		}
+	})
+	go periodicExecutor.Run(ctx)
+}
+
+func exitOnError(err error, context string) {
```

Comment:
The name suggests that that app exits if there is an error. Maybe change name to LogOnError

Reply:
log.D().Fatal exists the program